napari/utils/_tests/test_translations.py

```diff
@@ -0,0 +1,174 @@
+    p = subprocess.Popen(
+        [
+            "pybabel",
+            "compile",
+            "--domain=napari",
+            f"--dir={package_dir / 'locale'}",
+            f"--locale={TEST_LOCALE}",
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    stdout, stderr = p.communicate()
+    print(stdout, stderr)
+
+    print(
+        " ".join(
+            [
+                "pybabel",
+                "compile",
+                "--domain=napari",
+                f"--dir={package_dir / 'locale'}",
+                f"--locale={TEST_LOCALE}",
+            ]
+        )
+    )
+    # Load translator and force a locale for testing
+    translator._set_locale(TEST_LOCALE)
+    return translator.load()
+
+
+def test_get_display_name_valid():
+    assert _get_display_name("en", "en") == "English"
+    assert _get_display_name("en", "es") == "Inglés"
+    assert _get_display_name("en", "es_CO") == "Inglés"
+    assert _get_display_name("en", "fr") == "Anglais"
+    assert _get_display_name("es", "en") == "Spanish"
+    assert _get_display_name("fr", "en") == "French"
+
+
+def test_get_display_name_invalid():
+    assert _get_display_name("en", "foo") == "English"
+    assert _get_display_name("foo", "en") == "English"
+    assert _get_display_name("foo", "bar") == "English"
+
+
+def test_is_valid_locale_valid():
+    assert _is_valid_locale("en")
+    assert _is_valid_locale("es")
+    assert _is_valid_locale("es_CO")
+
+
+def test_is_valid_locale_invalid():
+    assert not _is_valid_locale("foo_SPAM")
+    assert not _is_valid_locale("bar")
+
+
+def test_run_process_and_parse(trans):
+    cmd = [
+        sys.executable,
+        translations.__file__,
+        "_get_installed_language_pack_locales",
+    ]
+    data, msg = _run_process_and_parse(cmd)
+    assert data == {'es_CO': 'napari_language_pack_es_CO'}
+    assert msg == ""
+
+
+def test_get_installed_packages_locales(trans):
+    data, msg = get_installed_packages_locales()
+    assert data == {'es_CO': 'napari_language_pack_es_CO'}
+    assert msg == ""
+
+
+def test_locale_valid_singular(trans):
+    # Test singular method
+    expected_result = "Más sobre napari"
+    result = trans.gettext("MORE ABOUT NAPARI")
+    assert result == expected_result
+
+    # Test singular method shorthand
+    result = trans._("MORE ABOUT NAPARI")
+    assert result == expected_result
+
+
+def test_locale_invalid():
+    translator._set_locale(TEST_LOCALE)
+    trans = translator.load()
+    result = trans._("BOO")
+    assert result == "BOO"
+
+
+def test_locale_n_runs(trans):
+    # Test plural method
+    n = 2
+    string = "MORE ABOUT NAPARI"
+    plural = "MORE ABOUT NAPARIS"
+    result = trans.ngettext(string, plural, n)
+    assert result == plural
+
+    # Test plural method shorthand
+    result = trans._n(string, plural, n)
+    assert result == plural
+
+
+@pytest.mark.xfail(reason="On ubuntu and python3.7 is failing")
```

Comment:
Do we know anything about these fails? Does it seem just on CI, or also locally. We could add a skip that was specific to 3.7 ubuntu, we've done things like that in other places

Reply:
>Do we know anything about these fails? Does it seem just on CI, or also locally. We could add a skip that was specific to 3.7 ubuntu, we've done things like that in other places

This was happenning only on linux and python3.7 on CI, I do not have a VM to test this myself locally. However it was a weird error since it was refering to missing methods on the gettext python module. But they should be there, so I was a bit puzzled.